pkg/helper/helper.go

```diff
@@ -0,0 +1,260 @@
+		}
+
+		return true, nil
+	})
+
+	AssertNoError(err, fmt.Sprintf("%s cluster CR deletion failed\n", name))
+}
+
+func ValidateSCCAdded(cs *client.Clients, ns, sa string) {
+	err := wait.Poll(config.APIRetry, config.APITimeout, func() (bool, error) {
+		privileged, err := GetPrivilegedSCC(cs)
+		if err != nil {
+			log.Printf("failed to get privileged scc: %s \n", err)
+			return false, err
+		}
+		log.Printf("... looking at %v", privileged.Users)
+
+		ctrlSA := fmt.Sprintf("system:serviceaccount:%s:%s", ns, sa)
+		return inList(privileged.Users, ctrlSA), nil
+	})
+	AssertNoError(err, fmt.Sprintf("failed to Add privilaged scc: %s\n", sa))
+}
+
+func ValidateSCCRemoved(cs *client.Clients, ns, sa string) {
+	err := wait.Poll(config.APIRetry, config.APITimeout, func() (bool, error) {
+		privileged, err := GetPrivilegedSCC(cs)
+		if err != nil {
+			log.Printf("failed to get privileged scc: %s \n", err)
+			return false, err
+		}
+
+		ctrlSA := fmt.Sprintf("system:serviceaccount:%s:%s", ns, sa)
+		return !inList(privileged.Users, ctrlSA), nil
+	})
+	AssertNoError(err, fmt.Sprintf("failed to Remove privilaged scc: %s\n", sa))
+}
+
+func inList(list []string, item string) bool {
+	for _, v := range list {
+		if v == item {
+			return true
+		}
+	}
+	return false
+}
+
+func ValidateDeployments(cs *client.Clients, ns string, deployments ...string) {
+
+	kc := cs.KubeClient.Kube
+	for _, d := range deployments {
+		err := WaitForDeployment(kc, ns,
+			d,
+			1,
+			config.APIRetry,
+			config.APITimeout,
+		)
+		AssertNoError(err, fmt.Sprintf("Deployments: %+v, failed to create\n", deployments))
+	}
+
+}
+
+func GetPrivilegedSCC(cs *client.Clients) (*secv1.SecurityContextConstraints, error) {
+	sec, err := secclient.NewForConfig(cs.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	return sec.SecurityContextConstraints().Get("privileged", metav1.GetOptions{})
+}
+
+func ValidateDeploymentDeletion(cs *client.Clients, ns string, deployments ...string) {
+
+	for _, d := range deployments {
+		err := WaitForDeploymentDeletion(cs, ns, d)
+		AssertNoError(err, fmt.Sprintf("Deployments: %+v, failed to delete\n", deployments))
+	}
+}
+
+func WaitForDeployment(kc kubernetes.Interface, namespace, name string, replicas int, retryInterval, timeout time.Duration) error {
+
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		deployment, err := kc.AppsV1().Deployments(namespace).Get(name, metav1.GetOptions{IncludeUninitialized: true})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Waiting for availability of %s deployment\n", name)
+				return false, nil
+			}
+			return false, err
+		}
+
+		if int(deployment.Status.AvailableReplicas) == replicas {
+			return true, nil
+		}
+		log.Printf("Waiting for full availability of %s deployment (%d/%d)\n", name, deployment.Status.AvailableReplicas, replicas)
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+	AssertNoError(err, fmt.Sprintf("Deployment failed\n"))
```

Comment:
This assert it useless, isn't it?

Reply:
ahh, yes I missed it. will update that Thanks.